### PR TITLE
bpo-44511: Improve the bytecode for class and mapping patterns

### PR DESCRIPTION
**Pull request by brandtbucher**
https://bugs.python.org/issue44511

### CODE REVIEW

**Comment by brandtbucher** [comment]
`asyncio` failure on Windows (`test_sendfile_close_peer_in_the_middle_of_receiving`) looks unrelated.

File: Lib/opcode.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def jabs_op(name, op):
 def_op('IS_OP', 117)
 def_op('CONTAINS_OP', 118)
 def_op('RERAISE', 119)
-
+def_op('PUSH_PEEK', 120)
```
Comment (markshannon) [inline Lib/opcode.py]: All instructions push their result. Could you rename this to `PEEK`, or `COPY`.
I'd prefer `COPY`.

File: Python/ceval.c [hunk 2]
```diff
@@ -4367,6 +4339,13 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
             DISPATCH();
         }
 
+        case TARGET(PUSH_PEEK): {
+            PyObject *peek = PEEK(oparg + 1);
```
Comment (markshannon) [inline Python/ceval.c]: I'd prefer `PEEK(oparg)`.
I realize that would make `oparg == 0` invalid, but counting from one is the convention when determining stack positions.

**Review by markshannon — COMMENTED** [review]
Looks like a nice improvement. A couple of minor comments

File: Doc/library/dis.rst [hunk 3]
```diff
@@ -754,15 +754,6 @@ iterations of the loop.
    .. versionadded:: 3.11
 
 
-.. opcode:: COPY_DICT_WITHOUT_KEYS
-
-   TOS is a tuple of mapping keys, and TOS1 is the match subject.  Replace TOS
-   with a :class:`dict` formed from the items of TOS1, but without any of the
-   keys in TOS.
-
-   .. versionadded:: 3.10
```
Comment (vstinner) [inline Doc/library/dis.rst:764]: Would it be possible to document bytecode changes in What's New in Python 3.11? There is a bytecode section:
https://docs.python.org/dev/whatsnew/3.11.html#cpython-bytecode-changes

File: Lib/opcode.py (hunk 1 above)
Comment (brandtbucher) [reply Lib/opcode.py]: Yeah, I wanted to just go with `PEEK`, but that conflicted with the `PEEK` macro in `ceval.c`.

I like `COPY` much better.

**Pull request merged**